### PR TITLE
Part 4: OpenIddict: Add API Client UI

### PR DESCRIPTION
**Pull request by Kukks**
Part 4: OpenIddict: Add API Client UI

### CODE REVIEW

**Comment by Kukks** [comment]
@NicolasDorier  I've rebased and fixed this PR, ready for the final push!

**Comment by NicolasDorier** [comment]
@Kukks integration test fails

File: BTCPayServer/Extensions/OpenIddictExtensions.cs [hunk 1]
```diff
@@ -10,29 +10,28 @@ namespace BTCPayServer
 {
     public static class OpenIddictExtensions
     {
-        private static SecurityKey _key = null;
```
Comment (NicolasDorier) [inline BTCPayServer/Extensions/OpenIddictExtensions.cs]: remove this cache

File: BTCPayServer/Pages/OpenId/Authorizations/Index.cshtml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                </table>
+            </div>
+        </div>
+    </div>
+</section>
+
+
+@functions{
+
+    [Authorize]
```
Comment (NicolasDorier) [inline BTCPayServer/Pages/OpenId/Authorizations/Index.cshtml]: The scheme must be explicitly set.

File: BTCPayServer/Pages/OpenId/Clients/Index.cshtml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                </table>
+            </div>
+        </div>
+    </div>
+</section>
+
+
+@functions{
+
+    [Authorize]
```
Comment (NicolasDorier) [inline BTCPayServer/Pages/OpenId/Clients/Index.cshtml]: The scheme must be explicitely set

File: BTCPayServer/Pages/OpenId/Tokens/Index.cshtml [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+                </table>
+            </div>
+        </div>
+    </div>
+</section>
+
+
+@functions{
+
+    [Authorize]
```
Comment (NicolasDorier) [inline BTCPayServer/Pages/OpenId/Tokens/Index.cshtml]: The scheme must be explicitely set

**Comment by NicolasDorier** [comment]
Add link info to explain wtf is hybrid etc... by linking to https://openid.net/specs/openid-connect-core-1_0.html

**Comment by NicolasDorier** [comment]
https://www.scottbrady91.com/OpenID-Connect/OpenID-Connect-Flows actually better

**Comment by NicolasDorier** [comment]
https://connect2id.com/learn/openid-connect

**Comment by NicolasDorier** [comment]
"Allow authentication through your client id and client secret. Note: requires your app to be confidential"

You mean confidential OR hybrid?

**Comment by Kukks** [comment]
> "Allow authentication through your client id and client secret. Note: requires your app to be confidential"
> 
> You mean confidential OR hybrid?

fixed

**Comment by NicolasDorier** [comment]
let's do this after we migrated to .netcoreapp3.0 openiddict is by far the biggest breaking change.

**Pull request closed without merging**